db.go

```diff
@@ -69,6 +94,20 @@ func (db *DB) String() string {
 	return b.String()
 }
 
+func (db *DB) Close() error {
+	db.closed.Store(true)
```

Comment:
```suggestion
	if db.closed.Swap(true) {
		return nil
	}
```

No need to close it again